game/src/core/components/PositionComponent.java

```diff
@@ -13,59 +13,63 @@
 
 import java.util.logging.Logger;
 
-/** A PositionComponent stores the associated entity's position in the level */
+/**
+ * The PositionComponent stores the {@link Point} of an Entity that represents the position of the
+ * associated entity in the level.
+ *
+ * <p>The {@link core.systems.DrawSystem} uses that position to draw the Entity at the correct
+ * location.
+ *
+ * <p>The {@link core.systems.VelocitySystem} will update the position values.
+ *
+ * <p>Other Systems will use the position for other calculations, like AI movement.
+ *
+ * @see Point
+ * @see core.systems.DrawSystem
+ * @see core.systems.VelocitySystem
+ */
 @DSLType(name = "position_component")
-public class PositionComponent extends Component {
+public final class PositionComponent extends Component {
 
-    private final Logger positionCompLogger = Logger.getLogger(this.getClass().getName());
-    private /*@DSLTypeMember(name="position")*/ Point position;
+    private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
+    private Point position;
 
     /**
-     * Creates a new PositionComponent at a given point.
-     *
-     * <p>Creates a new PositionComponent to store the associated entity's position in the level and
-     * add this component to the associated entity.
+     * Creates a new PositionComponent and adds it to the associated entity.
      *
      * <p>Sets the position of this entity to the given point.
      *
-     * @param entity associated entity
-     * @param point position of the entity
+     * @param entity The associated entity.
+     * @param position The position of the entity in the level.
      */
-    public PositionComponent(@DSLContextMember(name = "entity") Entity entity, Point point) {
+    public PositionComponent(final Entity entity, final Point position) {
         super(entity);
-
-        this.position = point;
+        this.position = position;
     }
 
     /**
-     * Creates a new PositionComponent at a given point.
-     *
-     * <p>Creates a new PositionComponent to store the associated entity's position in the level and
-     * add this component to the associated entity.
+     * Creates a new PositionComponent and adds it to the associated entity.
      *
      * <p>Sets the position of this entity to a point with the given x and y positions.
      *
      * @param entity associated entity
      * @param x x-position of the entity
      * @param y y-position of the entity
      */
-    public PositionComponent(@DSLContextMember(name = "entity") Entity entity, float x, float y) {
+    public PositionComponent(final Entity entity, final float x, final float y) {
```

Comment:
@cagix ich nutze diese Stelle mal als Beispiel für eine Fragen die bei mir aufgekommen  ist.

 `final` für call-by-value Parameter ist eigentlich unnütz. Wollen wir die dann weglassen?

Reply:
Ich finde es ziemlich grenzwertig, weil es etwas andeutet, was einfach nicht da ist. Wenn Du das Repo mal als Bewerbungsreferenz nehmen wollen würdest, würde ich sowas tunlichst unterlassen.

Reply:
> Wenn Du das Repo mal als Bewerbungsreferenz nehmen wollen würdest, würde ich sowas tunlichst unterlassen.

Drastische Worte (^^) aber klare Linie.


Nur um die Frage in den Raum zu werfen: Bin ich als zweit Semester dann verwirrt warum meine  `entity` final ist, mein `x` aber nicht?

Reply:
> Bin ich als zweit Semester dann verwirrt warum meine  entity final ist, mein x aber nicht?

möglicherweise. aber studium soll ja auch anregen zum nachforschen ;)